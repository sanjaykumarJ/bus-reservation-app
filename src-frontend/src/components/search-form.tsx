import React from 'react';
import ReactDatePicker from 'react-datepicker';
import "react-datepicker/dist/react-datepicker.css";
import Axios from 'axios';
import { BusListGrid, Record } from './bus-list-grid';

let axiosDefaults = require('axios/lib/defaults');
axiosDefaults.baseURL = 'http://localhost:8080/';

interface SearchFormState {
  value: string;
  departureCity: string;
  arrivalCity: string;
  startDate: Date;
  endDate: Date;
  records: Record[]
}
export class SearchForm extends React.Component<any, SearchFormState> {
  constructor(props: any) {
    super(props);
    this.state = {
      value: '',
      departureCity: '',
      arrivalCity: '',
      startDate: new Date(),
      endDate: new Date(),
      records: []
    };

    // this.handleChange = this.handleChange.bind(this);
    // this.handleSubmit = this.handleSubmit.bind(this);
    this.handleDepartureCityChange = this.handleDepartureCityChange.bind(this);
    this.handleArrivalCityChange = this.handleArrivalCityChange.bind(this);
    this.handleStartDateChange = this.handleStartDateChange.bind(this);
    this.handleEndDateChange = this.handleEndDateChange.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
  }

  handleDepartureCityChange(event: any) {
    this.setState({ departureCity: event.target.value });
  }

  handleArrivalCityChange(event: any) {
    this.setState({ arrivalCity: event.target.value });
  }

  handleStartDateChange(date: any) {
    this.setState({ startDate: date })
  }

  handleEndDateChange(date: any) {
    this.setState({ endDate: date })
  }

  async handleSearch(event: any) {
    if (this.state.departureCity && this.state.arrivalCity) {
      try {
        let response = await Axios.get('/querybuses', { params: { departureCity: this.state.departureCity, arrivalCity: this.state.arrivalCity } })
        console.log(response);
        this.setState({ records: response.data })
      } catch (e) {
        console.log(e.message);
      }
    } else {
      window.alert('Enter valid values in the required fields')
    }
  }

  render() {
    return (
      <div>
        <div className='flex-box'>
          <label aria-required={true}>
            {<p className='required-ash'>*</p>}Departure city:
            <input required type="text" value={this.state.departureCity} onChange={this.handleDepartureCityChange} />
          </label>
          <label>
            {<p className='required-ash'>*</p>}Arrival city:
            <input required type="text" value={this.state.arrivalCity} onChange={this.handleArrivalCityChange} />
          </label>
          <label>
            {<p className='required-ash'>*</p>}Departure date:
            <ReactDatePicker required selected={this.state.startDate}
              onChange={this.handleStartDateChange} />
          </label>
          <label>
            Arrival date:
            <ReactDatePicker selected={this.state.endDate}
              onChange={this.handleEndDateChange} />
          </label>
          <input type="button" onClick={this.handleSearch} value='Search Bus' />
        </div>
        {this.state.records.length > 0 && <BusListGrid records={this.state.records} />}

      </div>
    );
  }
}