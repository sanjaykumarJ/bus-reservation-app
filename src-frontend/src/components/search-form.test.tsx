import React from 'react';
import { render } from '@testing-library/react';
import { SearchForm } from './search-form';

const component = render(<SearchForm />)

describe('form component methods behaviour', () => {
    test('to match snapshot', () => {
        expect(component).toMatchSnapshot();
    })
})