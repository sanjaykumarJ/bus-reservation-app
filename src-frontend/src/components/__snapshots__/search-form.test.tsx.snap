// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form component methods behaviour to match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex-box"
      >
        <label
          aria-required="true"
        >
          <p
            class="required-ash"
          >
            *
          </p>
          Departure city:
          <input
            required=""
            type="text"
            value=""
          />
        </label>
        <label>
          <p
            class="required-ash"
          >
            *
          </p>
          Arrival city:
          <input
            required=""
            type="text"
            value=""
          />
        </label>
        <label>
          <p
            class="required-ash"
          >
            *
          </p>
          Departure date:
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class=""
                required=""
                type="text"
                value="07/19/2020"
              />
            </div>
          </div>
        </label>
        <label>
          Arrival date:
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class=""
                type="text"
                value="07/19/2020"
              />
            </div>
          </div>
        </label>
        <input
          type="button"
          value="Search Bus"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex-box"
    >
      <label
        aria-required="true"
      >
        <p
          class="required-ash"
        >
          *
        </p>
        Departure city:
        <input
          required=""
          type="text"
          value=""
        />
      </label>
      <label>
        <p
          class="required-ash"
        >
          *
        </p>
        Arrival city:
        <input
          required=""
          type="text"
          value=""
        />
      </label>
      <label>
        <p
          class="required-ash"
        >
          *
        </p>
        Departure date:
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class=""
              required=""
              type="text"
              value="07/19/2020"
            />
          </div>
        </div>
      </label>
      <label>
        Arrival date:
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class=""
              type="text"
              value="07/19/2020"
            />
          </div>
        </div>
      </label>
      <input
        type="button"
        value="Search Bus"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
